gleambatch.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import logging
+import random
+import subprocess
+import sys
+from pathlib import Path
+
+from epifor import Regions
+from epifor.common import die, yaml
+from epifor.data.batch import Batch
+from epifor.data.csse import CSSEData
+from epifor.data.foretold import FTData
+from epifor.gleam import GleamDef, Simulation
+
+log = logging.getLogger("gleambatch")
+
+
+def update_data(args):
+    "Fetch/update foretold and CSSE data (runs external scripts)"
+
+    with open(args.CONFIG_YAML, "rt") as f:
+        config = yaml.load(f)
+    Path(config["output_dir"]).expanduser().mkdir(exist_ok=True, parents=True)
+
+    if config["use_foretold"]:
+        log.info(f"Fetching Foretold data ...")
+        if config["foretold_channel"] == "SECRET":
+            die(
+                "`foretold_channel` in the config file is not set to non-default value."
+            )
+        cmd = [
+            "./fetch_foretold.py",
+            "-c",
+            config["foretold_channel"],
+            "-o",
+            config["foretold_file"],
+        ]
+        log.debug(f"Running {cmd!r}")
+        subprocess.run(cmd, check=True)
+    else:
+        log.info(f"Skipping Foretold update")
+
+    log.info(f"Fetching/updating CSSE John Hopkins data from github ...")
+    cmd = ["./fetch_csse.sh"]
+    log.debug(f"Running {cmd!r}")
+    subprocess.run(cmd, check=True)
+
+
+def estimate(batch, rs: Regions):
+    """
+    Create estimates and write them to `est` of all the regions.
+
+    Returns an updated GleamDef object.
+    """
+
+    # Fix any missing / inconsistent pops
+    rs.heuristic_set_pops()
+    rs.fix_min_pops()
+
+    # Load and apply CSSE
+    csse = CSSEData()
+    csse.load(batch.config["CSSE_dir"] + "/time_series_19-covid-{}.csv")
```

Comment:
If it was read from github directly, you could do:

```
from urllib.request import urlopen
result = urlopen('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv').read()
```

Reply:
Good point, added as a snippet to https://github.com/epidemics/covid/issues/233